R/DBI-package.R

```diff
@@ -17,3 +17,17 @@
 #'   Formal specification (currently work in progress and incomplete):
 #'   `vignette("spec", package = "DBI")`
 "_PACKAGE"
+
+require_arrow <- function() {
+  if (requireNamespace("arrow", quietly = TRUE)) {
+    return(invisible(TRUE))
+  }
+  stop("The arrow package is required for this functionality.")
+}
+
+if (requireNamespace("arrow", quietly = TRUE)) {
+  setOldClass(c("Table", "ArrowTabular", "ArrowObject", "R6"))
+  setOldClass(c("RecordBatch", "ArrowTabular", "ArrowObject", "R6"))
```

Comment:
Should this eventually be done by arrow itself?

Reply:
Not sure. Those are R6 objects and as such this is probably the S4 using package's responsibility, no?

Reply:
A crazy thought: since these are new generics, do we even need them to be S4, or could we move to S3?

Reply:
As discussed, I feel that the new generics should also be S4 for consistency reasons.